doc/SHOW_FORMATS_OPTION.md

```diff
@@ -0,0 +1,175 @@
+  - the value of this field is the origin of decision to skip, that's a label of branch in code that skipped the line (so you may check code in `loader.c`),
+- "uid", "gid", "gecos", "home", "shell" are for additional information about user (provided in some formats of hash files),
+  - they may be absent if they are empty,
+  - "gecos", "home", "shell" may be absent also if they have dummy value "/".
+
+"rowFormats" field contains a list of dictionaries with results of successful parsing of line by formats.
+
+Each dictionary in "rowFormats" list may have the following field/keys:
+- "label" is the name of format that may be used for `--format=` option,
+- "dynamic" is boolean value,
+  - it is true if format uses engine for dynamic formats,
+  - it is absent if it is false,
+- "prepareEqCiphertext" is boolean value,
+  - it is true if `prepare()` method of formats returned same ciphertext after processing (it may be interesting to developers of formats),
+  - it is absent if it is false,
+- "canonHash" is a list containing ciphertext in canonical form,
+  - it may contain multiple values for some formats of hash files (e.g. pwdump),
+  - canonical hash is almost unambiguous form of hash that allows john to load this hash with respective format without `--format=` option,
+    - there may be a few exceptional formats that have canonical form that cannot be distinguished from other formats,
+    - formats that are different implementations of same hash type have same canonical form usually (e.g. raw-md5 and raw-md5-opencl formats),
+  - cracked hashes are saved to .pot file in canonical form unless it is too long (see "truncated" field),
+- "truncHash" is a list like "canonHash" but contains shorter hash that would be used instead of canonical hash in .pot file (see "truncated" field),
+- "truncated" is boolean field that shows whether "canonHash" or "truncHash" is used for .pot file,
+  - it is absent when "canonHash" would be used in .pot file,
+  - it is true when "truncHash" would be used in .pot file,
+  - it may be true, while certain hash is short enough to be saved in canonical form, there is no "truncHash" field in this case.
+
+Example of hash transformation into canonical form:
+```
+$ cat 123456.pw
+e10adc3949ba59abbe56e057f20f883e
+```
+
+```
+$ john --format=raw-md5 123456.pw --show=formats
+..."rowFormats":[{"label":"Raw-MD5",...,"canonHash":["$dynamic_0$e10adc3949ba59abbe56e057f20f883e"]}]}]
+```
+
+```
+$ john --format=raw-md5 123456.pw --pot=123456.pot
+[...]
+123456           (?)
+[...]
+```
+
+```
+$ cat 123456.pot
+$dynamic_0$e10adc3949ba59abbe56e057f20f883e:123456
+```
+
+## Interaction with other options
+
+Formats to be tried may be limited with `--format=` option.
+- 1 exact format may be specified (e.g. `--format=raw-md5`),
+  - a disabled dynamic format may be enabled this way for temporary use,
+- other formats will not be checked and reported, it may be useful because `--show=formats` may be slow.
```

Comment:
Just FWIW one could mention format wildcards around here, although they are (I think) mentioned somewhere else. Examples: `--format=*crypt`, `--format=mssql*`, `--format=raw*256`. They may come handy using this.

Reply:
I intended to write that, but it does not work really.

Reply:
hm, raw*256 works; I tried --format=*-ztex and john tells `Unknown ciphertext format name requested`

Reply:
For `--format=*-ztex` to work you need to have a ZTEX-enabled build. Other than that, it really should work.

Reply:
We're no longer registering the ZTEX formats with `--show` in order to avoid initializing the boards unnecessarily and allow for `--show` to work concurrently with actual use of those boards.

Reply:
Ah, yes. I have considered doing something similar for OpenCL.